cli/test/proxy.test.ts

```diff
@@ -0,0 +1,103 @@
+import { describe, test, expect, vi, beforeEach } from 'vitest';
+import { setGlobalDispatcher, ProxyAgent } from 'undici';
+import { HttpsProxyAgent } from 'https-proxy-agent';
+
+// Mocks the actual used undici modules
+// The tests check if those have been configured
+// correctly after starting the cli with or without
+// a proxy
+vi.mock('undici', () => ({
+  setGlobalDispatcher: vi.fn(),
+  ProxyAgent: vi.fn().mockImplementation((opts) => ({
+    uri: opts.uri,
+  })),
+}));
+
+// Mocks https-proxy-agent used inside the
+// connectrpc platform client which will be
+// injected whenever a proxy was configured
+vi.mock('https-proxy-agent', () => ({
+  HttpsProxyAgent: vi.fn().mockImplementation((proxyUrl) => ({
+    proxyUrl,
+  })),
+}));
+
+// ensure all env variables and spies are cleaned up after each test
+const prepare = () => {
+  // resets e.g. ProxyAgent mock state between describes
+  vi.resetModules();
+  vi.resetAllMocks();
+
+  delete process.env.HTTPS_PROXY;
+  delete process.env.HTTP_PROXY;
+};
+
+beforeEach(prepare);
+
+describe('fetch when the HTTP_PROXY variable is set', () => {
+  test('when using HTTP_PROXY the ProxyAgent for native fetch was configured', async () => {
```

Comment:
`test` should read like english, e.g., `test('that when...')`.

We don't tend to use `it`, but if it were `it`: `it('works when...')`